### ISSUE TITLE
No checks for undefined properties causing rooms to die

### ISSUE THREAD

**Issue by webremote** [ScreepsOCS/screeps.behaviour-action-pattern#510]
If we would implement MORE checks against undefined, rooms wouldn't die:

```
TypeError: Cannot read property 'data' of undefined
    at validateRunning (task.guard:139:26)
    at Array.forEach (native)
    at Object.handleCreepDied (task.guard:146:28)
    at Creep.died.on.name (task.guard:15:43)
    at handlers.slice.forEach.h (global:25:57)
    at Array.forEach (native)
    at LiteEvent.trigger (global:25:44)
    at register (population:163:28)```

Fix:

Line 135 at task.guard.js

```
let creep = Game.creeps[o];
                // invalidate old creeps for predicted spawning
                // TODO: better distance calculation
                let prediction;
                if (creep === undefined) {
                    return;
                }
```
Almost killed my rooms....

**Comment by dkluempers** [comment]
I tried that but it did  not clear the problem.  I deleted the flags and it cleared.  I will try it again.

**Comment by cyberblast** [comment]
fixed in #511

**Comment by karlthepagan** [comment]
looks good in dev
